Use `/Fe<file>` (single arg) syntax when generating `tasks.json` to support older versions of `cl.exe`

Bug type: Language Service

<!-- Prior to creating a bug report, please review:
📝 Existing issues at https://github.com/Microsoft/vscode-cpptools/issues
📜 Our documentation at https://code.visualstudio.com/docs/languages/cpp
📙 FAQs at https://code.visualstudio.com/docs/cpp/faq-cpp
-->


**Describe the bug**
- OS and Version: Windows 10 20H2
- VS Code Version: 1.62.2
- C/C++ Extension Version: 1.7.1
- I installed Windows SDK 7.1 on Windows 10. I launch the developer command prompt, typing cl gives "Microsoft (R) 32-bit C/C++ Optimizing Compiler Version 16.00.40219.01 for 80x86" and start code from the prompt. Whatever the project is single or multi projects, cl.exe is not detected.
- If I set the compilerpath manually in c_cpp_properties.json, the includepath is not set. If I manually set the includepath, the build failed in terminal: 
```
Starting build...
cl.exe /Zi /EHsc /nologo /Fe: C:\Home\Public\test\main.exe C:\Home\Public\test\main.c
cl : Command line warning D9024 : 
unrecognized source file type 'C:\Home\Public\test\main.exe', object file assumed


main.c

LINK : fatal error LNK1181: cannot open input file 'C:\Home\Public\test\main.exe'

Build finished successfully.
```


**Expected behavior**
Correct detection of cl.exe and paths. Successful build and coherent output message, why is the build finished successfully?

- Configurations in `c_cpp_properties.json` after manual settings of compilerpath and includepath
```
{
    "configurations": [
        {
            "name": "Win32",
            "includePath": [
                "${workspaceFolder}/**",
                "C:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\VC\\include"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE"
            ],
            "cStandard": "c89",
            "cppStandard": "c++98",
            "intelliSenseMode": "windows-msvc-x86",
            "compilerPath": "C:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\VC\\bin\\cl.exe"
        }
    ],
    "version": 4
}
```

- generated tasks.json
```
{
			"type": "cppbuild",
			"label": "C/C++: cl.exe build active file",
			"command": "cl.exe",
			"args": [
				"/Zi",
				"/EHsc",
				"/nologo",
				"/Fe:",
				"${fileDirname}\\${fileBasenameNoExtension}.exe",
				"${file}"
			],
			"options": {
				"cwd": "${fileDirname}"
			},
			"problemMatcher": [
				"$msCompile"
			],
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"detail": "compiler: cl.exe"
		}
```

--- a/Extension/src/LanguageServer/cppBuildTaskProvider.ts
+++ b/Extension/src/LanguageServer/cppBuildTaskProvider.ts
@@ -169,7 +169,7 @@ export class CppBuildTaskProvider implements TaskProvider {
             const taskLabel: string = ((appendSourceToName && !compilerPathBase.startsWith(ext.configPrefix)) ?
                 ext.configPrefix : "") + compilerPathBase + " " + localize("build_active_file", "build active file");
             const programName: string = util.defaultExePath();
-            let args: string[] = isCl ? ['/Zi', '/EHsc', '/nologo', '/Fe:', programName, '${file}'] : ['-fdiagnostics-color=always', '-g', '${file}', '-o', programName];
+            let args: string[] = isCl ? ['/Zi', '/EHsc', '/nologo', `/Fe${programName}`, '${file}'] : ['-fdiagnostics-color=always', '-g', '${file}', '-o', programName];
             if (compilerArgs && compilerArgs.length > 0) {
                 args = args.concat(compilerArgs);
             }

--- a/Extension/src/common.ts
+++ b/Extension/src/common.ts
@@ -328,7 +328,11 @@ export function defaultExePath(): string {
 }
 
 export function findExePathInArgs(args: string[]): string | undefined {
-    return args.find((arg: string, index: number) => (arg.includes(".exe") || (index > 0 && args[index - 1] === "-o")));
+    const exePath: string | undefined = args.find((arg: string, index: number) => (arg.includes(".exe") || (index > 0 && args[index - 1] === "-o")));
+    if (exePath?.startsWith("/Fe")) {
+        return exePath.substring(3);
+    }
+    return exePath;
 }
 
 // Pass in 'arrayResults' if a string[] result is possible and a delimited string result is undesirable.

